repose-aggregator/components/services/opentracing-service/opentracing-service-impl/src/main/java/org/openrepose/core/services/opentracing/OpenTracingServiceImpl.java

```diff
@@ -0,0 +1,269 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.core.services.opentracing;
+
+import com.uber.jaeger.Configuration;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+import org.openrepose.commons.config.manager.UpdateListener;
+import org.openrepose.core.service.opentracing.config.OpenTracingConfig;
+import org.openrepose.core.services.config.ConfigurationService;
+import org.openrepose.core.services.healthcheck.HealthCheckService;
+import org.openrepose.core.services.healthcheck.HealthCheckServiceProxy;
+import org.openrepose.core.services.healthcheck.Severity;
+import org.openrepose.core.services.opentracing.interceptors.JaegerRequestInterceptor;
+import org.openrepose.core.services.opentracing.interceptors.JaegerResponseInterceptor;
+import org.openrepose.core.services.opentracing.interceptors.RequestInterceptor;
+import org.openrepose.core.services.opentracing.interceptors.ResponseInterceptor;
+import org.slf4j.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ * OpenTracingService - service that integrates OpenTracing standards into Repose
+ */
+@Named
+public class OpenTracingServiceImpl implements OpenTracingService {
+
+    public static final String DEFAULT_CONFIG_NAME = "opentracing.cfg.xml";
+
+    private static final Logger LOG = org.slf4j.LoggerFactory.getLogger(OpenTracingServiceImpl.class);
+    private static final String OPENTRACING_SERVICE_REPORT = "OpenTracingServiceReport";
+
+    private final ConfigurationService configurationService;
+    private final ConfigurationListener configurationListener;
+    private final HealthCheckServiceProxy healthCheckServiceProxy;
+
+    private boolean isEnabled = false;
+
+    private boolean initialized = false;
+
+    private String serviceName = null;
+
+    private RequestInterceptor requestInterceptor;
+    private ResponseInterceptor responseInterceptor;
+
+
+    /**
+     * Initial constructor
+     * @param configurationService
+     * @param healthCheckService
+     */
+    @Inject
+    public OpenTracingServiceImpl(
+        ConfigurationService configurationService,
+        HealthCheckService healthCheckService
+    ) {
+        LOG.debug("Creating New OpenTracing Service");
+
+        this.configurationService = configurationService;
+        this.healthCheckServiceProxy = healthCheckService.register();
+        this.configurationListener = new ConfigurationListener();
+    }
+
+    @PostConstruct
+    public void init() {
+        LOG.debug("Initializing OpenTracingService");
+
+        healthCheckServiceProxy.reportIssue(
```

Comment:
Health Check Service isn't really doing what we want it to, so we don't use it much. Feel free to not do all this.

Reply:
If I don't have it, will it not be able to start my service?  I've noticed that the healthcheck checks if anything that's registered is broken and then won't start repose...